src/components/atom/button/Button.tsx

```diff
@@ -0,0 +1,83 @@
+import cn from '@/utils/cn';
+import { cva, type VariantProps } from 'class-variance-authority';
+import Image from 'next/image';
+
+type RoundedSize = 'none' | 'sm' | 'md' | 'lg' | 'xl' | 'full';
+
+const buttonVariants = cva(
+  'w-full min-w-[158px] h-[43px] font-sans font-semibold flex items-center justify-center hover:brightness-90',
+  {
+    variants: {
+      variant: {
+        primary: 'bg-primary-black text-primary-white',
+        outlinedBlack:
+          'border-2 border-primary-black text-primary-black hover:border-opacity-70',
+        outlinedWhite: 'border-2 border-primary-white text-primary-white',
+        gray: 'bg-primary-gray text-primary-white',
+        cyan: 'bg-secondary-cyan text-primary-white',
+        red: 'bg-secondary-red text-primary-white',
+      },
+      rounded: {
+        none: 'rounded-none',
+        sm: 'rounded-sm',
+        md: 'rounded-md',
+        lg: 'rounded-lg',
+        xl: 'rounded-xl',
+        full: 'rounded-full',
+      },
+    },
+    defaultVariants: {
+      variant: 'primary',
+      rounded: 'none',
+    },
+  },
+);
+
+type ButtonProps = {
+  children: React.ReactNode;
+  rounded?: RoundedSize;
+  height?: string;
+  fontSize?: string;
+  iconSrc?: string;
+  iconSize?: number;
+  iconPosition?: 'left' | 'right';
+} & React.ButtonHTMLAttributes<HTMLButtonElement> &
+  VariantProps<typeof buttonVariants>;
+
+export default function Button({
+  children,
+  variant,
+  rounded,
+  iconSrc,
+  iconSize = 14,
+  iconPosition = 'left',
+  className,
+  ...props
+}: ButtonProps) {
+  return (
+    <button
+      className={cn(buttonVariants({ variant, rounded }), className)}
+      {...props}
+    >
+      {iconSrc && iconPosition === 'left' && (
+        <Image
+          src={iconSrc}
+          alt='icon'
+          width={iconSize}
+          height={iconSize}
+          className='mr-2'
+        />
+      )}
+      {children}
+      {iconSrc && iconPosition === 'right' && (
+        <Image
+          src={iconSrc}
+          alt='icon'
+          width={iconSize}
+          height={iconSize}
+          className='ml-3'
+        />
+      )}
```

Comment:
```suggestion
      const icon = iconSrc ? (
  <Image
    src={iconSrc}
    alt="icon"
    width={iconSize}
    height={iconSize}
    className={iconPosition === "left" ? "mr-2" : "ml-3"}
  />
) : null;

{iconPosition === "left" && icon}
{children}
{iconPosition === "right" && icon}
```
지금 코드에서는 왼쪽과 오른쪽에 대해 각각 조건문에서 처리하고 있어서 코드가 길어지는 것 같아요
어차피 `iconSrc` 가 없으면 아이콘을 렌더링할 필요가 없으니
변수를 만들어 `null`로 처리한 후, 한 번만 체크하고 위치만 다르게 설정하면 더 간결해질 것 같은 느낌이 드네요~

Reply:
좋은 방법인 것 같아요!!
다른 분들이 읽었을 때 조건같은 부분이 한 번에 이해될 수 있게 작성해봤는데 제안해주신 방법이 확실히 더 좋은 것 같습니다!! 수정해보겠습니다!!
늦게 답변드려서 죄송합니다. 😂